R/get_carto.R

```diff
@@ -0,0 +1,553 @@
+#' Get a Carto basemap
+#'
+#' \code{get_carto} accesses a tile server for Carto basemaps and
+#' downloads/stitches map tiles/formats a map image.
+#'
+#' @param bbox a bounding box in the format c(lowerleftlon, lowerleftlat,
+#'   upperrightlon, upperrightlat).
+#' @param zoom a zoom level
+#' @param maptype light_all, dark_all, light_nolabels, light_only_labels, dark_nolabels, dark_only_labels,
+#' rastertiles/voyager, rastertiles/voyager_nolabels, rastertiles/voyager_only_labels, or rastertiles/voyager_labels_under.
+#' @param crop crop raw map tiles to specified bounding box. if FALSE, the
+#'   resulting map will more than cover the bounding box specified.
+#' @param messaging turn messaging on/off
+#' @param urlonly return url only
+#' @param color color or black-and-white (use force = TRUE if you've already
+#'   downloaded the images)
+#' @param force if the map is on file, should a new map be looked up?
+#' @param where where should the file drawer be located (without terminating
+#'   "/")
+#' @param https if TRUE, queries an https endpoint so that web traffic between
+#'   you and the tile server is ecrypted using SSL.
```

Comment:
This should be an opt-out. There's no good reason not to use HTTPS when it's available.